cpp/src/gandiva/function_registry_string.cc

```diff
@@ -68,7 +68,20 @@ std::vector<NativeFunction> GetStringFunctionRegistry() {
 
       NativeFunction("like", {}, DataTypeVector{utf8(), utf8()}, boolean(),
                      kResultNullIfNull, "gdv_fn_like_utf8_utf8",
-                     NativeFunction::kNeedsFunctionHolder)};
+                     NativeFunction::kNeedsFunctionHolder),
+      NativeFunction("substr", {"substring"}, DataTypeVector{utf8(), int64(), int64()},
```

Comment:
These are non-obvious - can you please add comments for the params/functions ?

Reply:
done